provider-anynines/internal/controller/servicebinding/servicebinding.go

```diff
@@ -262,12 +263,44 @@ func (c external) GetServiceInstanceManagedResource(ctx context.Context, sb v1.S
 	return instances.ToServiceInstance(sb.Name)
 }
 
-// initializeServiceBindingStatus initializes InstanceID, ServiceID and PlanID in the status
-// if not set.
+// GetServiceBindingSecret retrieves the servicebinding secret with postfix '-creds'.
+func (c external) GetServiceBindingSecret(ctx context.Context, sb v1.ServiceBinding) (*corev1.Secret, error) {
+	secret := &corev1.Secret{}
+
+	err := c.kube.Get(ctx, types.NamespacedName{
+		Name:      sb.Name + "-creds",
+		Namespace: sb.Labels[constants.LabelKeyClaimNamespace],
+	}, secret)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to get ServiceBinding secret: %w",
+			err)
+	}
+
+	return secret, nil
+}
+
+// initializeServiceBindingStatus initializes InstanceID, ServiceID, PlanID and ConnectionDetails
+// in the status if not set.
 func (c *external) initializeServiceBindingStatus(ctx context.Context, sb *v1.ServiceBinding) error {
```

Comment:
This function does not exceed the cyclic complexity limit, so the decision is more stylistic. However, readability could be improved by creating single-responsibility functions to handle `initializeServiceBindingStatus`, `initializeConnectionDetails`, and `initializeInstanceFields`. This would enhance the code's readability and maintainability.